src/driver/mysql/MysqlDriver.ts

```diff
@@ -471,6 +473,9 @@ export class MysqlDriver implements Driver {
 
         } else if (columnMetadata.type === "simple-json") {
             value = DateUtils.stringToSimpleJson(value);
+
+        } else if (columnMetadata.type === "enum" && !isNaN(value)) {
```

Comment:
what if value was defined as `"1"` and user expects numbers?

Reply:
I don't think we'll be able to make "heterogeneous" way possible to work. The only option I see is to support string-only or number-only. If enum was defined in a   "heterogeneous" way - we either throw error either enforce strings.

(to understand if its string or numeric based we do `Object.keys(enum).every(key => typeof enum[key] === "number")`)

Reply:
Yeah I decided to force all string numbers to classic javascript numbers, so values of this enum will be transformed to number after select from databases:
```typescript
enum TestEnum {
    NO = "0",
    YES = "1"
}
```
But in line you are highlighted we can check if exists numeric enum value (`Object.keys(enum).some(key => enum[key] === parseInt(value))`) and convert it only if exists. After that change the only one edge case will be enum like this:
```typescript
enum TestEnum {
    YES1 = 1
    YES = "1"
}
```
And this looks like really edge case.

Reply:
Right, last case is really edge case and I think its okay if we don't cover it. But first case should work properly (convert "0" to string, not number).

Reply:
Updated src and added test enum with numeric values as string.